### DISCUSSION TITLE
Searching with romaji

### DISCUSSION THREAD

**Discussion by rasmus-kirk** [obfusk/jiten#123] [Ideas]
Hi, not a bug or anything but it would be nice to be able to search with romaji. For example:
`jiten -v jmdict --max 1 --word neko --lang romaji`
would return the same as
`jiten -v jmdict --max 1 --word cat`
Probably not the best syntax for it, but I hope you get the idea. Also, not related but worth a mention, there is a lot of empty lines at the bottom of a request:
```
[user@nixos:~]$ jiten -v jmdict --max 1 --word cat
DB v11 up to date.
query: +w cat

猫
ねこ | ネコ
ねꜜこ
[eng]
= cat (esp. the domestic cat, Felis catus)
= shamisen
= geisha
= wheelbarrow
= clay bed-warmer
= bottom | submissive partner of a homosexual relationship
--> noun (common) (futsuumeishi) | abbreviation | word usually written
    using kana alone | colloquialism
see 猫車 | 猫火鉢
seq# 1467640, freq# 2201, jlpt N5, prio; 1




[user@nixos:~]$ 
``` 
I remove it with `jiten query |  head -n -3` so it's not a big deal or anything, just a minor annoyance.

**Comment by obfusk** [comment]
> there is a lot of empty lines at the bottom

I know. But it never bothered me enough to fix :sweat_smile:

One of those is "my fault": I just fixed that in commit f3bef9ddc7d917a408ccedd485186ee1afad5365.

One of those is caused by a library I'm using: I worked around that in the same commit.

One of those is caused by the pager (`less -FR`): I can't fix that (but if you don't want a pager, `jiten --colour ... | cat` or `jiten --pager cat --colour ...` will).

And I think one of those is caused by the default nixos setting for `$PS1` :)

**Comment by obfusk** [comment]
> it would be nice to be able to search with romaji

FWIW: the web/android interface already supports converting romaji to kana.

I'll look into adding support for that to the CLI as well.

**Comment by obfusk** [comment]
Of course you can already use a Japanese input method like `fcitx`+`mozc` (`nixos.fcitx-engines.mozc` on nixos) that allows you to input kana by typing in romaji.

**Comment by obfusk** [comment]
I added support for `-h/--hiragana` and `-k/--katakana` in commit db336ef195d6fbc0e2518b79df5ab2af6946e376. For example:

```
$ jiten -v jmdict -m1 --word --hiragana neko
$ jiten -v jmdict -m1 --word --katakana terebi
```

Testing & feedback is always appreciated :)
